cnf-tests/testsuites/e2esuite/dpdk/dpdk.go

```diff
@@ -173,6 +179,99 @@ var _ = Describe("dpdk", func() {
 		}
 	})
 
+	Context("vhostnet", func() {
+		var policyHasVhostnet bool
+		execute.BeforeAll(func() {
+			if !discovery.Enabled() {
+				CleanSriov()
+				createSriovPolicyAndNetworkDPDKOnlyWithVhost()
+			} else {
+				sriovNetworkNodePolicyList := &sriovv1.SriovNetworkNodePolicyList{}
+				err := client.Client.List(context.TODO(), sriovNetworkNodePolicyList)
+				Expect(err).ToNot(HaveOccurred())
+				for _, policy := range sriovNetworkNodePolicyList.Items {
+					if policy.Spec.ResourceName == dpdkResourceName && policy.Spec.NeedVhostNet {
+						policyHasVhostnet = true
+					}
+				}
+			}
+		})
+
+		BeforeEach(func() {
+			if discovery.Enabled() && !policyHasVhostnet {
+				Skip("Missing SriovNetworkNodePolicy with NeedVhostNet enabled")
+			}
+		})
+		Context("Client should be able to forward packets", func() {
+			It("Should be able to transmit packets", func() {
+				var out string
+				var err error
+
+				command := CREATE_TAP_DEVICE_COMMAND + `
+				dpdk-testpmd --vdev net_tap0,iface=tap23 --no-pci -- -ia --forward-mode txonly
+				sleep INF
+				`
+				txDpdkWorkloadPod, err := createDPDKWorkload(nodeSelector,
+					command,
+					false,
+					[]corev1.Capability{"NET_ADMIN", "MKNOD"},
+				)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Parsing output from the client DPDK application")
+				Eventually(func() string {
+					out, err = pods.GetLog(txDpdkWorkloadPod)
+					Expect(err).ToNot(HaveOccurred())
+					return out
+				}, 2*time.Minute, 1*time.Second).Should(ContainSubstring(LOG_ENTRY),
+					"Cannot find accumulated statistics")
+				By("Checking the tx output from the client DPDK application")
+				checkTxOnly(out)
+			})
+		})
+
+		Context("Server should be able to receive packets and forward to tap device", func() {
+			It("Should be able to transmit packets", func() {
+				var err error
+				var out string
+				// --stats-period is used to keep the command alive once the pod is started
+				serverCommand := fmt.Sprintf(`
+%s
+dpdk-testpmd --vdev net_tap0,iface=tap23 -w ${PCIDEVICE_OPENSHIFT_IO_%s} -- --stats-period 5
+sleep INF
+				`, CREATE_TAP_DEVICE_COMMAND, strings.ToUpper(dpdkResourceName))
+				dpdkWorkloadPod, err := createDPDKWorkload(nodeSelector, serverCommand, false, []corev1.Capability{"NET_ADMIN", "MKNOD"})
+				Expect(err).ToNot(HaveOccurred())
+
+				mac, err := getDeviceMac(dpdkWorkloadPod, "net1")
```

Comment:
This is not right you will not see the device if it's a intel nic.

better is to configure in the pod the mac that you want.

example:
```
apiVersion: v1
kind: Pod
metadata:
  annotations:
    k8s.v1.cni.cncf.io/networks: '[
      {
       "name": "dpdk-network-3",
       "mac": "60:00:00:00:00:01",
       "namespace": "seba"
      },
      {
       "name": "dpdk-network-4",
       "mac": "60:00:00:00:00:02",
       "namespace": "seba"
      }
    ]'
    cpu-load-balancing.crio.io: "true"
  labels:
    app: dpdk
  name: testpmd
  namespace: seba
spec:
....
```

Reply:
Done